rclcpp/include/rclcpp/publisher.hpp

```diff
@@ -164,6 +168,9 @@ class Publisher : public PublisherBase
     }
   }
 
+  [[deprecated(
+    "publishing an unique_ptr is prefered when using intra process communication."
+    " If using a shared_ptr, use publish(*msg).")]]
```

Comment:
This suggestion won't work because the only other signature for an `rcl_serialized_message_t` is `publish(const rcl_serialized_message_t *)`.

If we had a `publish(const rcl_serialized_message_t &)` then this message would make more sense. I think that would be a good thing to have, and in which case we could deprecate `publish(const rcl_serialized_message_t *)` too, that way were only ever dealing with `unique_ptr` and references, which is ideal in my opinion.